### ISSUE TITLE
Segmentation fault with impl Iterator

### ISSUE THREAD

**Issue by bytesnake** [zshipko/rust-kv#8]
Hey there,
I get a segmentation fault when I return the `Iterator`, but not the corresponding `Cursor`, like so:
```rust
pub fn publications(&self) -> impl Iterator<Item = (u64, Publication)> {
        self.txn.read_cursor(&self.bucket).unwrap().iter()
            .map(|(a,b)| (u64::from(a), b.inner().unwrap().to_serde()))
}
```
this results in an `Segmentation Fault`. My solution was to wrap the `Cursor` in a seperate Iterator struct and implementing the next function manually.

**Comment by zshipko** [comment]
Thanks! 

I don't really have enough information to be able to replicate this, but I've just made some changes to `cursor.rs`. Let me know if using the kv git repo in your Cargo.toml fixes the issue.

I've also added an iter test: https://github.com/zshipko/rust-kv/commit/881155243d8eb0720728dffac9291be918225e48#diff-c5fc4beef0f03493c2e9a9c876073f0fR104, feel free to make a PR to update this test case to be more similar to your example.
